src/app/rpmostree-builtin-kargs.cxx

```diff
@@ -266,6 +271,8 @@ rpmostree_builtin_kargs (int            argc,
   g_variant_dict_insert (&dict, "reboot", "b", opt_reboot);
   g_variant_dict_insert (&dict, "initiating-command-line", "s", invocation->command_line);
   g_variant_dict_insert (&dict, "lock-finalization", "b", opt_lock_finalization);
+  g_variant_dict_insert (&dict, "append-if-missing", "^as", opt_kernel_append_if_missing_strings);
+  g_variant_dict_insert (&dict, "detele-if-present", "^as", opt_kernel_delete_if_present_strings);
```

Comment:
We should only do this if the options were actually provided. So we should move up the `if-statement`s added below here instead. E.g.

```c
if (opt_kernel_... && *opt_kernel_...)
  g_variant_dict_insert (...)
```

Reply:
Ok, I think I've done it but please check if I understood right.